src/pages/home/ReportScreen.tsx

```diff
@@ -489,12 +522,46 @@ function ReportScreen({
                 !isLoading &&
                 !userLeavingStatus) ||
             shouldHideReport ||
-            (!!reportIDFromParams && !ReportUtils.isValidReportIDFromPath(reportIDFromParams)),
-        [report, reportMetadata, isLoading, shouldHideReport, isOptimisticDelete, userLeavingStatus, reportIDFromParams],
+            (!!reportIDFromRoute && !ReportUtils.isValidReportIDFromPath(reportIDFromRoute)),
+        [report, reportMetadata, isLoading, shouldHideReport, isOptimisticDelete, userLeavingStatus, reportIDFromRoute],
     );
 
     const actionListValue = useMemo((): ActionListContextType => ({flatListRef, scrollPosition, setScrollPosition}), [flatListRef, scrollPosition, setScrollPosition]);
 
+    // This helps in tracking from the moment 'route' triggers useMemo until isLoadingInitialReportActions becomes true. It prevents blinking when loading reportActions from cache.
+    useEffect(() => {
+        InteractionManager.runAfterInteractions(() => {
+            setLinkingToMessage(false);
+        });
+    }, [reportMetadata?.isLoadingInitialReportActions]);
+
+    const onLinkPress = () => {
+        Navigation.setParams({reportActionID: ''});
+        fetchReport();
+    };
+
+    const isLinkedReportActionDeleted = useMemo(() => {
+        if (!reportActionIDFromRoute || !sortedAllReportActions) {
+            return false;
+        }
+        const action = sortedAllReportActions.find((item) => item.reportActionID === reportActionIDFromRoute);
+        return action && ReportActionsUtils.isDeletedAction(action);
+    }, [reportActionIDFromRoute, sortedAllReportActions]);
+
+    if (isLinkedReportActionDeleted ?? (!shouldShowSkeleton && reportActionIDFromRoute && reportActions?.length === 0 && !isLinkingToMessage)) {
```

Comment:
We should have considered whisper actions which the user does not have access to here. This later caused #41370